OpenRA.Mods.Common/Widgets/ExpSliderWidget.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	public class ExpSliderWidget : SliderWidget
+	{
+		public ExpSliderWidget() { }
+		public ExpSliderWidget(ExpSliderWidget other) : base(other) { }
+
+		float ExpFromLinear(float x)
+		{
+			return (float)(x == MinimumValue ? MinimumValue : 1.0e-3 * Math.Exp(6.908 * x)).Clamp(MinimumValue, MaximumValue);
+		}
+
+		float LinearFromExp(float x)
+		{
+			return (float)(x == MinimumValue ? MinimumValue : Math.Log(x / 1.0e-3) / 6.908).Clamp(MinimumValue, MaximumValue);
+		}
+
+		protected override float ValueFromPx(int x)
+		{
+			return ExpFromLinear(MinimumValue + (MaximumValue - MinimumValue) * (x - 0.5f * RenderBounds.Height) / (RenderBounds.Width - RenderBounds.Height));
+		}
+
+		protected override int PxFromValue(float x)
+		{
+			return (int)(0.5f * RenderBounds.Height + (RenderBounds.Width - RenderBounds.Height) * (LinearFromExp(x) - MinimumValue) / (MaximumValue - MinimumValue));
```

Comment:
Why multiply by 0.5 when you divide by 2 in the Draw function? Not a deal breaker, but I'd personally prefer division by 2.

Reply:
This is the exact copy of the same methods as in SliderWidget:
https://github.com/OpenRA/OpenRA/blob/fadcfa0828bee9d00fcfdaccca92970be70a4926/OpenRA.Mods.Common/Widgets/SliderWidget.cs#L87-95